js/client/src/components/app/map/Map.js

```diff
@@ -151,22 +207,37 @@ const ConnectedMap = ({
       }
     };
 
-    map.on("draw.create", () => {
-      drawnData();
-      map.scrollZoom.disable();
-    });
-    map.on("draw.update", () => drawnData());
-    map.on("draw.delete", () => {
-      handleDrawing(false);
-      sideBar[0].classList.remove("--container-open");
-      map.on("click", "places", layerClick);
-      var drawPolygon = document.getElementsByClassName(
-        "mapbox-gl-draw_polygon"
-      );
-      drawPolygon[0].disabled = false;
-      drawPolygon[0].classList.remove("disabled-button");
-      map.scrollZoom.enable();
-    });
+    const zipStatics = async (zip) => {
+      try {
+        const response = await axios
+        .get(`${API_URL}/api/citation/draw/zip`, {
+          params: {
+            zip: zip,
+          },
+        })
+        setData(response.data)
+        getPolygonData(zip);
+        handleDrawing(true)
+        sideBar[0].classList.add("--container-open");
+        map.setLayoutProperty("heatmap", 'visibility', 'visible')
+        map.setLayoutProperty("places", 'visibility', 'visible')
+      } catch (err) {
+        console.log(err)
+      }
```

Comment:
same thing here, I'm thinking we maybe able to create a top level HoC to handle all exceptions instead try/catching every http request.

Reply:
another approach is wrap the axios to our custom axios library and add handler in the abstracted layer, is also a good approach to only handle http exceptions.

Reply:
example for HoC: [https://codeburst.io/catching-exceptions-using-higher-order-components-in-react-16-b8a401853a10](https://codeburst.io/catching-exceptions-using-higher-order-components-in-react-16-b8a401853a10)